Source/WebCore/page/scrolling/ScrollSnapOffsetsInfo.h

```diff
@@ -38,12 +38,15 @@ namespace WebCore {
 class ScrollableArea;
 class RenderBox;
 class RenderStyle;
+class Element;
 
 template <typename T>
 struct SnapOffset {
     T offset;
     ScrollSnapStop stop;
     bool hasSnapAreaLargerThanViewport;
+    unsigned snapTargetId;
```

Comment:
Rather than storing this extra ID, and the associated (and tricky to maintain) renderer -> ID map, can we just compute the snap offset for the focus thing and send it along as the "preferred snap offset" or "preferred snap index"?

Any way that doesn't involve maintaining a RenderBox -> ID map would be preferable.

Reply:
The issue with storing a snap index/offset is that it will change across layout, so when `updateSnapOffsetsForScrollableArea` is called, we have no way to figure out what the corresponding index/offset is. Instead of having an extra map, would it be preferable to replace the `HashSet<const RenderBox*>` already in `RenderView` with a set of structs which hold the `RenderBox` and id?

Reply:
Actually this is not a great solution as we would have to do the work ourselves to maintain the set, due to the ids. This could of course work, but since we are maintaining the map in the same way as the set (which was on RenderView already), could you explain why this isn't preferable?